demonstrations/tutorial_noisy_circuit_optimization.py

```diff
@@ -0,0 +1,519 @@
+
+Background
+----------
+
+Quantum pure-state simulators are great and readily available in 
+a number of quantum software packages.
+They allow us to experiment, prototype, test, and validate algorithms
+and research ideas---up to a certain number of qubits, at least.
+
+But present-day hardware is not ideal. We're forced to confront 
+decoherence, bit flips, amplitude damping, and so on. Does the 
+presence of noise in near-term devices impact their use in, 
+for example, 
+:doc:`variational quantum algorithms </glossary/variational_circuit>`?
+Won't our models, trained so carefully in simulators, fall apart 
+when we run on noisy devices?
+
+In fact, there is some optimism that variational algorithms may be
+the best type of algorithms on near-term devices, and could
+have an in-built adaptability to noise that more rigid 
+textbook algorithms do not possess. 
+Variational algorithms are somewhat robust against the fact 
+that the device they are run on may not be ideal. Being variational
+in nature, they can be tuned to "work around" noise to some extent. 
+
+Quantum machine learning leverages a lot of tools from its 
+classical counterpart. Fortunately, there is great evidence that
+machine learning algorithms can not only be robust to noise, but 
+can even benefit from it! 
+Examples include the use of 
+`reduced-precision arithmetic in deep learning <https://dl.acm.org/doi/abs/10.5555/3045118.3045303>`_,
+the strong performance of
+`stochastic gradient descent <https://en.wikipedia.org/wiki/Stochastic_gradient_descent>`_, 
+and the use of "dropout" noise to 
+`prevent overfitting <https://www.cs.toronto.edu/~hinton/absps/JMLRdropout.pdf>`_. 
+
+With this evidence to motivate us, we can still hope to find, 
+extract, and work with the underlying quantum "signal" 
+that is influenced by a device's inherent noise. 
+
+Noisy circuits: creating a Bell state
+-------------------------------------
+
+Let's consider a simple quantum circuit which performs a standard
+quantum information task: the creation of an entangled
+state and the measurement of a 
+`Bell inequality <https://en.wikipedia.org/wiki/Bell%27s_theorem>`_
+(also known as the 
+`CHSH inequality <https://en.wikipedia.org/wiki/CHSH_inequality>`_).
+
+Since we'll be dealing with noise, we'll need to use a simulator
+that supports noise and density-state representations of quantum
+states (in contrast to many simulators, which use a pure-state
+representation). 
+
+Fortunately, `Cirq <https://cirq.readthedocs.io>`_ provides mixed-state
+simulators and noisy operations natively, so we can use the 
+PennyLane-Cirq plugin to carry out our noisy simulations.
+"""
+
+import pennylane as qml
+from pennylane import numpy as np
+import matplotlib.pyplot as plt
+
+dev = qml.device("cirq.mixedsimulator", wires=2)
+
+# CHSH observables
+A1 = qml.PauliZ(0)
+A2 = qml.PauliX(0)
+B1 = qml.Hermitian(np.array([[1, 1], [1, -1]]) / np.sqrt(2), wires=1)
```

Comment:
Isn't this just the following?
```suggestion
B1 = qml.Hadamard(1)
```

Reply:
Yes, but I put it this way to show the similarity to the next obserable

Reply:
btw would be good if operators composed nicely, e.g., I really wanted to write `B2 = qml.PauliY @ qml.Hermitian @ qml.PauliY` (or whatever the actual relation is)

Reply:
I think this works as is? Just need to specify the `matrix` property:
![image](https://user-images.githubusercontent.com/2959003/83474746-df4e7a00-a4cb-11ea-866c-949712f05468.png)